test/common/http/conn_manager_impl_test_2.cc

```diff
@@ -2667,16 +2667,17 @@ TEST_F(HttpConnectionManagerImplTest, TestUpstreamRequestHeadersSize) {
       .WillOnce(Return(FilterHeadersStatus::StopIteration));
   EXPECT_CALL(*decoder_filters_[0], decodeComplete());
 
-  std::shared_ptr<NiceMock<Upstream::MockHostDescription>> host_{
-      new NiceMock<Upstream::MockHostDescription>()};
-  filter_callbacks_.upstreamHost(host_);
+  std::shared_ptr<Upstream::MockThreadLocalCluster> cluster =
```

Comment:
actually this is a case where `auto` is fine as it's clear from the RHS what the type has to be